model-optimizer/extensions/middle/ReverseV2ToReverseSequence.py

```diff
@@ -32,29 +38,55 @@ def pattern():
 
     def replace_pattern(self, graph: Graph, match: dict):
         reverse = match['reverse']
-        input_data_shape = reverse.in_node(0).shape
-
-        if len(input_data_shape) == 1:
-            raise Error('Reverse operation name = {} is\'t supported because of 1D input.'.format(reverse.name))
+        reverse_name = reverse.soft_get('name', reverse.id)
 
         assert reverse.in_port(1).disconnected()
 
-        seq_axis = reverse['axis']
-        # We need to choose arbitrary batch_axis != sequence_axis
-        batch_axis = int(not seq_axis)
+        # add new dimension as batch
+        unsq_node = create_op_node_with_second_input(graph, Unsqueeze, int64_array([0]),
+                                                     {'name': reverse_name+"/Unsqueeze"})
+        reverse.in_port(0).get_source().connect(unsq_node.in_port(0))
+        # set layout as correct to avoid adding Transpose nodes
+        unsq_node.in_port(1).__setattr__('input_permutation', None)
+        mark_input_as_in_correct_layout(unsq_node, 0)
+        mark_output_as_in_correct_layout(unsq_node, 0)
+        new_in = unsq_node.out_port(0)
+        batch_axis = int64_array([0])
+        seq_axis = reverse['axis'] + 1 if reverse['axis'] >= 0 else reverse['axis']  # add 1 for newly added dimension
 
         # 1. For ReverseSequence 1-port input is seq_lengths => create this input node
-        seq_lengths = np.ones(input_data_shape[batch_axis]) * input_data_shape[seq_axis]
-
         reverse_name = reverse.soft_get('name',  reverse.id)
         rename_node(reverse, reverse_name + '/to_delete')
+
+        shape_node = Shape(graph, {'name': reverse_name + "/shape"}).create_node()
+        new_in.connect(shape_node.in_port(0))
+        mark_input_as_in_correct_layout(shape_node, 0)
```

Comment:
Why we have so many calls to mark_***_as_in_correct_layout here? Dont think this is necessary and a good code style.

Reply:
removed